okapi-operation/src/axum_integration/mod.rs

```diff
@@ -35,8 +39,21 @@ use crate::*;
         )
     )
 )]
-pub async fn serve_openapi_spec(spec: State<OpenApi>) -> Json<OpenApi> {
-    Json(spec.0)
+pub async fn serve_openapi_spec(spec: State<OpenApi>, headers: HeaderMap) -> Response {
+    match headers.get("Accept").and_then(|h| h.to_str().ok()) {
+        Some("yaml") => AxumYaml(spec.0).into_response(),
+        Some("json") => Json(spec.0).into_response(),
+        Some(_) => {
+            let status = StatusCode::BAD_REQUEST;
+            let headers = [(
+                header::CONTENT_TYPE,
+                HeaderValue::from_static("text/plain; charset=utf-8"),
+            )];
+            let err = format!("Bad Accept header value, should be [yaml|json]");
```

Comment:
Maybe mention that Accept header should be empty?

Reply:
Fixed